api_v3/lib/types/EntryServerNode/base/filters/KalturaEntryServerNodeFilter.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * @package api
+ * @subpackage filters
+ */
+class KalturaEntryServerNodeFilter extends KalturaEntryServerNodeBaseFilter
+{
+	/**
+	 * @return baseObjectFilter
+	 */
+	protected function getCoreFilter()
+	{
+		return new EntryServerNodeFilter();
+	}
+
+	/**
+	 * @param KalturaFilterPager $pager
+	 * @param KalturaDetachedResponseProfile $responseProfile
+	 * @return KalturaListResponse
+	 * @throws KalturaAPIException
+	 */
+	public function getListResponse(KalturaFilterPager $pager, KalturaDetachedResponseProfile $responseProfile = null)
+	{
+		if ($this->entryIdEqual == null && $this->entryIdNotIn == null && $this->entryIdIn == null &&
+			$this->idEqual == null && $this->idNotIn == null && $this->idIn == null &&
+			$this->serverTypeEqual == null )
+		{
+			throw new KalturaAPIException(KalturaErrors::MUST_FILTER_ON_ENTRY_AND_SERVER_TYPE);
+		}
+
+		if($this->entryIdEqual != null)
+		{
+			$entry = entryPeer::retrieveByPK($this->entryIdEqual);
+			if(!$entry)
+				throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $this->entryIdEqual);
+		}
+
+		if($this->entryIdIn != null)
+		{
+			$entryArray = entryPeer::retrieveByPKs($this->entryIdIn);
+			if(!$entryArray)
```

Comment:
here you should check count on entryArray?
Also what about cases where some of the entries in the entryIdIn did not return in the entryArray don't you want to remove them from the filter? 
or simply do implode(",", $entryArray)